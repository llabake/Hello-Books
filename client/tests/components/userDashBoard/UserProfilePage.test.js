import React from 'react';

import {
  UserProfilePage,
  mapStateToProps,
  mapDispatchToProps
} from '../../../components/userDashBoard/UserProfilePage';

import adminData from '../../mocks/userData';
import userData from '../../mocks/userData';


function setup() {
  const props = {
    dispatch: jest.fn(),
    user: adminData.admin,
    getUserProfile: jest.fn(),
  };

  const wrapper = shallow(<UserProfilePage {...props} />);

  return {
    wrapper,
    props,
  };
}

describe('UserProfilePage page', () => {
  test('should render and not crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  })

  test('should call component did mount', () => {
    let spy;
    afterEach(() => {
      spy.mockClear();
    });
    spy = jest.spyOn(UserProfilePage.prototype, 'componentDidMount')
    const { wrapper, props } = setup();
    expect(wrapper).toBeDefined();
    expect(UserProfilePage.prototype.componentDidMount).toHaveBeenCalledTimes(1)
  })

  test('should return an object with props required by component from store', () => {
    const store = {
      user: {}
    };
    expect(mapStateToProps({ userReducer: {} })).toHaveProperty('user');
  });

  test('mapDispatchToProps', () => {
    const dispatch = jest.fn();

    expect(mapDispatchToProps(dispatch)).toHaveProperty('getUserProfile');

    const { getUserProfile } = mapDispatchToProps(dispatch);
    getUserProfile();
    expect(dispatch).toHaveBeenCalled();
  });

}) 