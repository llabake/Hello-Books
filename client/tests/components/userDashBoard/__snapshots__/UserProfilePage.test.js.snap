// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfilePage page should render and not crash 1`] = `
<div>
  <div
    className="row "
  >
    <Connect(UserProfileSidePanel)
      user={
        Object {
          "active": true,
          "email": "oreitanoyebola@gmail.com",
          "id": 4,
          "role": "admin",
          "username": "keinzy",
        }
      }
    />
    <div
      className="col s12 m8"
    >
      <Connect(UserBorrowedBookList) />
    </div>
  </div>
</div>
`;
